system-addon/content-src/components/TopSites/TopSites.jsx

```diff
@@ -117,25 +117,24 @@ class TopSitesPerfTimer extends React.Component {
   }
 
   /**
-   * Call the given callback when the subsequent animation frame
-   * (not the upcoming one) paints.
+   * Call the given callback after the upcoming frame paints.  We're using
+   * a Promise rather than a setTimeout or double rFA, as Promise resolution
+   * is faster, as of this writing (Firefox 57 Nightly) - see #3105 for
```

Comment:
I'll update the stuff referring to Promises to talk about setTimeout (and maybe postMessage).